lib/static/components/details.jsx

```diff
@@ -1,89 +1,81 @@
     };
 
-    _getContent() {
-        const content = this.props.content;
+    const getContent = () => {
+        const {content} = props;
 
         return isFunction(content) ? content() : content;
-    }
+    };
 
-    _renderContent() {
-        if (!this.state.isOpened) {
+    const renderContent = () => {
+        if (!isOpened) {
             return null;
         }
 
-        const children = this.props.asHtml ? null : this._getContent();
-        const extraProps = this.props.asHtml ? {dangerouslySetInnerHTML: {__html: this._getContent()}} : {};
+        const children = props.asHtml ? null : getContent();
+        const extraProps = props.asHtml ? {dangerouslySetInnerHTML: {__html: getContent()}} : {};
 
         return <div className='details__content' {...extraProps}>
             {children}
         </div>;
-    }
+    };
 
-    render() {
-        const {type, title, content, extendClassNames} = this.props;
-        const className = classNames(
-            'details',
-            extendClassNames
-        );
+    const {title, content, extendClassNames} = props;
+    const className = classNames(
+        'details',
+        extendClassNames
+    );
 
-        return (
-            isEmpty(content) && !isFunction(content) ? (
-                <div className={className}>
-                    {title}
-                </div>
-            ) : (
-                <Disclosure className={className} onUpdate={this.handleClick}
-                    size='l'>
-                    <Disclosure.Summary>
-                        {(props, defaultButton) => (
-                            <div className={classNames(className, 'details__summary')} aria-controls={props.ariaControls} onClick={props.onClick} id={props.id}>
-                                <div className="details__expand-button" onClick={this.stopPropagation}>
-                                    {defaultButton}
-                                </div>
-                                {title}
+    const {measure} = useContext(MeasurementContext);
+    useLayoutEffect(() => {
+        measure?.();
+    }, [isOpened]);
+
+    return (
+        isEmpty(content) && !isFunction(content) ? (
+            <div className={className}>
+                {title}
+            </div>
+        ) : (
+            <Disclosure className={className} onUpdate={handleClick} size='l' expanded={isOpened}>
+                <Disclosure.Summary>
+                    {(props, defaultButton) => (
+                        <div className={classNames(className, 'details__summary')} aria-controls={props.ariaControls} onClick={props.onClick} id={props.id}>
+                            <div className="details__expand-button" onClick={e => e.stopPropagation()}>
+                                {defaultButton}
                             </div>
-                        )}
-                    </Disclosure.Summary>
-                    {type === 'image' ? this._renderContent() :
```

Comment:
why don't you need this condition now?

Reply:
We wrap components in Card directly where we need it and pass already wrapped components as children.

The Details component shouldn't be aware of anything regarding its children IMO. It shouldn't care what it will render, it just receives `ReactNode` and renders it as-is. That makes logic a lot easier to understand and reduces complexity. That's from the software design point of view.

Another reason why I got rid of it, because it cause another visual glitch — upon collapsing Meta/History, you'd first see a frame where Disclosure contains empty card and only afterwards it collapses completely.